src/lib/Mapper/PjaxResponseMainContentMapper.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace EzSystems\HybridPlatformUi\Mapper;
+
+use EzSystems\HybridPlatformUi\Components\MainContent;
+use Symfony\Component\HttpFoundation\Response;
+
+class PjaxResponseMainContentMapper implements MainContentMapper
+{
+    /**
+     * @var \EzSystems\HybridPlatformUi\Components\MainContent
+     */
+    private $mainContent;
+
+    public function __construct(MainContent $mainContent)
+    {
+        $this->mainContent = $mainContent;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Response $response
+     *
+     * @return \EzSystems\HybridPlatformUi\Components\MainContent
+     * @throws \Exception
+     */
+    public function map($response)
+    {
+        if (!$response instanceof Response) {
+            throw new \InvalidArgumentException('Expected a \Symfony\Component\HttpFoundation\Response');
+        }
+
+        $responseContent = $response->getContent();
+
+        $errorHandling = libxml_use_internal_errors(true);
+
+        $doc = new \DOMDocument();
+        if (!$doc->loadHTML($responseContent)) {
+            $errors = libxml_get_errors();
+            libxml_use_internal_errors($errorHandling);
+            throw new \Exception(
+                "Error(s) occurred parsing the PJAX response:\n" .
+                implode("\n", $errors)
+            );
+        }
+
+        $xpath = new \DOMXPath($doc);
+        $nodeList = $xpath->query('//div[@data-name="title"]');
+        $title = $nodeList[0]->nodeValue;
+        $contentNodeList = $xpath->query('//div[@data-name="html"]//section[contains(concat(" ", normalize-space(@class), " "), " ez-serverside-content ")]');
+
+        $contentNode = $contentNodeList[0];
+        $attributes = 'ez-view-serversideview';
+        foreach (explode(' ', $contentNode->getAttribute('class')) as $classAttribute) {
+            if ($classAttribute !== 'ez-view-serversideview') {
+                $attributes .= " $classAttribute";
+            }
+        }
+        $content =
+            '<div class="'.$attributes.'">' .
+            $this->innerHtml($contentNode) .
```

Comment:
this feels a bit complicated and scary but I like the `innerHtml` method ;)

Joke aside, why not just wrap the content of `div[@data-name="html"]` in `<div class="ez-view-serversideview"></div>` ?